tests/types/ember-concurrency-test.ts

```diff
@@ -486,6 +767,228 @@ module('unit tests', () => {
     }
   });
 
+  test('EncapsulatedTaskProperty', async () => {
+    // @ts-expect-error
+    new EncapsulatedTaskProperty(); // EncapsulatedTaskProperty cannot be constructed
+
+    // @ts-expect-error
+    class Foo extends EncapsulatedTaskProperty<any, any[], {}> {} // EncapsulatedTaskProperty cannot be subclassed
+
+    let tp!: EncapsulatedTaskProperty<string, [boolean, number?], { foo: string }>;
+
+    // We inherited from ComputedProperty to make Ember.get work, but a TP is
+    // not actually a CP so we need to make sure CP methods are not callable
+
+    expect(tp.volatile).toBeNever();
+    expect(tp.readOnly).toBeNever();
+    expect(tp.property).toBeNever();
+    expect(tp.meta).toBeNever();
+
+    // @ts-expect-error
+    tp.volatile();
+
+    // @ts-expect-error
+    tp.readOnly();
+
+    // @ts-expect-error
+    tp.property();
+
+    // @ts-expect-error
+    tp.meta();
+
+    expect(tp.on).toBeCallableWith();
+    expect(tp.on).toBeCallableWith('init');
+    expect(tp.on).toBeCallableWith('init', 'didInsertElement');
+    expect(tp.on).parameters.toEqualTypeOf<string[]>();
+    expect(tp.on).returns.toEqualTypeOf<typeof tp>();
+
+    // @ts-expect-error
+    tp.on(false);
+
+    // @ts-expect-error
+    tp.on('init', false);
+
+    expect(tp.cancelOn).toBeCallableWith();
+    expect(tp.cancelOn).toBeCallableWith('init');
+    expect(tp.cancelOn).toBeCallableWith('init', 'didInsertElement');
+    expect(tp.cancelOn).parameters.toEqualTypeOf<string[]>();
+    expect(tp.cancelOn).returns.toEqualTypeOf<typeof tp>();
+
+    // @ts-expect-error
+    tp.cancelOn(false);
+
+    // @ts-expect-error
+    tp.cancelOn('init', false);
+
+    expect(tp.restartable).toBeCallableWith();
+    expect(tp.restartable).parameters.toEqualTypeOf<[]>();
+    expect(tp.restartable).returns.toEqualTypeOf<typeof tp>();
```

Comment:
Here and elsewhere where you’re writing it this way, can you not just write it like this?

```suggestion
    expect(tp.restartable).returns.toEqualTypeOf(tp);
```